### ISSUE TITLE
[JENKINS-32686] Limit Scan by Job Label, Job Type, Name Etc.

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/build-failure-analyzer-plugin#287]
<p>We know that certain types of failures are only going to occur on certain kinds of jobs.  Especially when a mulit-line can is required, it is ideal if we an limit what jobs are scanned for a particular cause.  Obvious ways to limit a job scanning are as follows:</p>

<ul>
	<li>By job label.</li>
	<li>By Name (regex pattern match against name)</li>
	<li>By job Type (maven, free style, etc.)</li>
	<li>By what folder or view the job appears in.</li>
</ul>

<p>Some of these ideas are harder to implement than others and probably deserve their own ticket.  The first two are easy and likely give the most ROI.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kbaltrinic.png" title="kbaltrinic's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kbaltrinic">kbaltrinic</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32686" target="_blank">Limit Scan by Job Label, Job Type, Name Etc.</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/t_westling.jpg" title="t_westling's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=t_westling">t_westling</a>
<li><b>status</b>: Open
<li><b>priority</b>: Major
<li><b>component(s)</b>: build-failure-analyzer-plugin
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 1
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-12-07
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>We know that certain types of failures are only going to occur on certain kinds of jobs.  Especially when a mulit-line can is required, it is ideal if we an limit what jobs are scanned for a particular cause.  Obvious ways to limit a job scanning are as follows:</p>

<ul>
	<li>By job label.</li>
	<li>By Name (regex pattern match against name)</li>
	<li>By job Type (maven, free style, etc.)</li>
	<li>By what folder or view the job appears in.</li>
</ul>

<p>Some of these ideas are harder to implement than others and probably deserve their own ticket.  The first two are easy and likely give the most ROI.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/t_westling.jpg" title="t_westling's avatar" /> t_westling:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32686?focusedId=247195&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-247195">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I assume that the main driver for this idea is to spend less time and resources on scanning?
Sounds like a good idea and I agree, the obvious place to start is with the easy implementations e.g. job name.</p>
</pre>
</details>

</ul>
</details>

<p>I assume that the main driver for this idea is to spend less time and resources on scanning?<br/>
Sounds like a good idea and I agree, the obvious place to start is with the easy implementations e.g. job name.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kbaltrinic.png" title="kbaltrinic's avatar" /> kbaltrinic:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32686?focusedId=247298&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-247298">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Perhaps, but the biggest bang for the buck is in the label one I think.  We have windows slaves and linux slaves building very different things and they don't share much in the way of know build failures.</p>
</pre>
</details>

</ul>
</details>

<p>Perhaps, but the biggest bang for the buck is in the label one I think.  We have windows slaves and linux slaves building very different things and they don't share much in the way of know build failures.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kbaltrinic.png" title="kbaltrinic's avatar" /> kbaltrinic:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32686?focusedId=247299&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-247299">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Some more thinking on the label one.  It seems to me that if you allow KB entries to be labeled much the same way as slave nodes are labeled, then you should be able to use the same logic that determines "can this job run on this slave?" to determine "should this job's logs be evaluated against this indication?"  Thus I can label a known docker error with the "docker" label and it will only be run against those jobs that would have run on my 'docker' slaves, for example.  </p>

<p>As for the saving processor time, yes, this is particularly important when doing multi-line indications.  if I can tailor those to a very specific set of jobs, it would be great.</p>
</pre>
</details>

</ul>
</details>

<p>Some more thinking on the label one.  It seems to me that if you allow KB entries to be labeled much the same way as slave nodes are labeled, then you should be able to use the same logic that determines "can this job run on this slave?" to determine "should this job's logs be evaluated against this indication?"  Thus I can label a known docker error with the "docker" label and it will only be run against those jobs that would have run on my 'docker' slaves, for example.  </p>

<p>As for the saving processor time, yes, this is particularly important when doing multi-line indications.  if I can tailor those to a very specific set of jobs, it would be great.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/t_westling.jpg" title="t_westling's avatar" /> t_westling:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32686?focusedId=247377&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-247377">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Maybe the BFA categories can be reused for this? By default, each project would be scanned for everything,
but a project setting could tell BFA to only scan for failures that are part of certain categories.
I feel like a new "label" concept in parallel with categories could be confusing, e.g. the failure cause
"can't run docker for some reason" with category "docker" and label "docker".</p>
</pre>
</details>

</ul>
</details>

<p>Maybe the BFA categories can be reused for this? By default, each project would be scanned for everything,<br/>
but a project setting could tell BFA to only scan for failures that are part of certain categories.<br/>
I feel like a new "label" concept in parallel with categories could be confusing, e.g. the failure cause<br/>
"can't run docker for some reason" with category "docker" and label "docker".</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kbaltrinic.png" title="kbaltrinic's avatar" /> kbaltrinic:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32686?focusedId=257021&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-257021">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sorry I dropped this thread but I would say I am cool on this solution.  This means we would need to update every single job config and we have a lot more jobs that build failure entries.  And the jobs are naturally categorized already by labels.  </p>

<p>In addition, we heavily use the categories and feed this data into or own analytic platform where we track what kinds of failures occur how frequently, which are trending, etc.  So for us this would sort -of dual-purpose the category field and dirty up our analytics.</p>
</pre>
</details>

</ul>
</details>

<p>Sorry I dropped this thread but I would say I am cool on this solution.  This means we would need to update every single job config and we have a lot more jobs that build failure entries.  And the jobs are naturally categorized already by labels.  </p>

<p>In addition, we heavily use the categories and feed this data into or own analytic platform where we track what kinds of failures occur how frequently, which are trending, etc.  So for us this would sort -of dual-purpose the category field and dirty up our analytics.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kbaltrinic.png" title="kbaltrinic's avatar" /> kbaltrinic:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32686?focusedId=275779&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-275779">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Any more motion on this?  Would be super happy just to have the job-name-regex implemented.</p>
</pre>
</details>

</ul>
</details>

<p>Any more motion on this?  Would be super happy just to have the job-name-regex implemented.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ssbarnea.png" title="ssbarnea's avatar" /> ssbarnea:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32686?focusedId=301260&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-301260">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I am interested about any way for enabling BFA to work on only a limited number of jobs. We have really fat Jenkins instances and enabling BFA on some of them is a always-postponed due to risks of affecting other jobs.</p>

<p>Without a configurable way to limit BFA on some of the jobs we are stuck. A configurable way to do this would allow us to tune in-out its usage without having to enable/disable the entire plugin.</p>
</pre>
</details>

</ul>
</details>

<p>I am interested about any way for enabling BFA to work on only a limited number of jobs. We have really fat Jenkins instances and enabling BFA on some of them is a always-postponed due to risks of affecting other jobs.</p>

<p>Without a configurable way to limit BFA on some of the jobs we are stuck. A configurable way to do this would allow us to tune in-out its usage without having to enable/disable the entire plugin.</p>
